### PR TITLE
Sync phone-number's expected error messages to canonical data

### PR DESCRIPTION
**Pull request by BNAndras**

### CODE REVIEW

File: exercises/practice/phone-number/.meta/lib/example.dart [hunk 1]
```diff
@@ -13,27 +13,40 @@ class PhoneNumber {
       phoneNumber = phoneNumber.replaceAll(match.group(0)!, "");
     });
 
-    if (phoneNumber.contains(RegExp(r'[a-zA-Z]'))) throw FormatException('letters not permitted');
+    if (phoneNumber.contains(RegExp(r'[a-zA-Z]')))
+      throw FormatException('letters not permitted');
```
Comment (Stargator) [inline exercises/practice/phone-number/.meta/lib/example.dart:17]: I don't think a new line is needed here.

File: exercises/practice/phone-number/.meta/lib/example.dart [hunk 2]
```diff
@@ -13,27 +13,40 @@ class PhoneNumber {
...
 
-    if (phoneNumber.contains(RegExp(r'!|@'))) throw FormatException('punctuations not permitted');
+    if (phoneNumber.contains(RegExp(r'!|@')))
+      throw FormatException('punctuations not permitted');
```
Comment (Stargator) [inline exercises/practice/phone-number/.meta/lib/example.dart:20]: I don't think a new line is needed here.

File: exercises/practice/phone-number/.meta/lib/example.dart [hunk 3]
```diff
@@ -13,27 +13,40 @@ class PhoneNumber {
...
 
-    if (onlyDigits.length < 10) throw FormatException('incorrect number of digits');
+    if (onlyDigits.length < 10)
+      throw FormatException('must not be fewer than 10 digits');
```
Comment (Stargator) [inline exercises/practice/phone-number/.meta/lib/example.dart:23]: I don't think a new line is needed here. But the other change is fine, I'll need to check the canonical-json

File: exercises/practice/phone-number/.meta/lib/example.dart [hunk 4]
```diff
@@ -13,27 +13,40 @@ class PhoneNumber {
...
 
-    if (onlyDigits.length > 11) throw FormatException('more than 11 digits');
+    if (onlyDigits.length > 11)
+      throw FormatException('must not be greater than 11 digits');
```
Comment (Stargator) [inline exercises/practice/phone-number/.meta/lib/example.dart:26]: I don't think a new line is needed here. I'll check the canonical data for the other change.

File: exercises/practice/phone-number/.meta/lib/example.dart [hunk 5]
```diff
@@ -13,27 +13,40 @@ class PhoneNumber {
...
+      if (int.parse(onlyDigits[0]) != 1)
+        throw FormatException('11 digits must start with 1');
+      if (int.parse(onlyDigits[1]) == 0)
+        throw FormatException('area code cannot start with zero');
+      if (int.parse(onlyDigits[1]) == 1)
+        throw FormatException('area code cannot start with one');
+      if (int.parse(onlyDigits[4]) == 0)
+        throw FormatException('exchange code cannot start with zero');
+      if (int.parse(onlyDigits[4]) == 1)
+        throw FormatException('exchange code cannot start with one');
```
Comment (Stargator) [inline exercises/practice/phone-number/.meta/lib/example.dart:38]: Considering the nested nature of this section, I would suggest using braces for these if statements.

File: exercises/practice/phone-number/.meta/lib/example.dart [hunk 6]
```diff
@@ -13,27 +13,40 @@ class PhoneNumber {
...
-      if (int.parse(onlyDigits[3]) == 0) throw FormatException('exchange code cannot start with zero');
-      if (int.parse(onlyDigits[3]) == 1) throw FormatException('exchange code cannot start with one');
+      if (int.parse(onlyDigits[0]) == 0)
+        throw FormatException('area code cannot start with zero');
+      if (int.parse(onlyDigits[0]) == 1)
+        throw FormatException('area code cannot start with one');
+      if (int.parse(onlyDigits[3]) == 0)
+        throw FormatException('exchange code cannot start with zero');
+      if (int.parse(onlyDigits[3]) == 1)
+        throw FormatException('exchange code cannot start with one');
```
Comment (Stargator) [inline exercises/practice/phone-number/.meta/lib/example.dart:49]: Considering the nested nature of this section, I would suggest using braces for these if statements.

**Comment by Stargator** [comment]
Thanks for the changes, @BNAndras. I'll need to catch up to problem-specifications repo and verify the changes. But I appreciate the contribution.

File: exercises/practice/phone-number/.meta/lib/example.dart (hunk 1 above)
Comment (BNAndras) [reply exercises/practice/phone-number/.meta/lib/example.dart:17]: That appears to have been done by the VS Code Dart extension, not `dart format`. I'll revert the newline change globally

File: exercises/practice/phone-number/.meta/lib/example.dart (hunk 5 above)
Comment (BNAndras) [reply exercises/practice/phone-number/.meta/lib/example.dart:38]: To clarify, did you want me to add those braces or revert the changes per my other comment?

Comment (Stargator) [reply exercises/practice/phone-number/.meta/lib/example.dart:38]: Add the braces. Something like:

```dart
if (condition) {
  throw Exception();
}

if (conditionB) {
  throw Exception();
}
```

**Comment by Stargator** [comment]
Look over the canonical-data and yeah just add braces where I noted and revert the other new lines, then you should be clear to go forward.

**Review by Stargator — APPROVED** [review]

**Review by Stargator — APPROVED** [review]

**Pull request merged**